python/src/iceberg/transforms.py

```diff
@@ -0,0 +1,520 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import base64
+import math
+import re
+import struct
+from datetime import datetime, timedelta
+from decimal import Decimal
+
+import mmh3  # type: ignore
+import pytz
+
+from .types import (
+    BinaryType,
+    DateType,
+    DecimalType,
+    DoubleType,
+    FixedType,
+    FloatType,
+    IntegerType,
+    LongType,
+    StringType,
+    TimestampType,
+    TimestamptzType,
+    TimeType,
+    Type,
+    UUIDType,
+)
+
+
+class Transform(object):
+    """
+    Transform base class for concrete transforms. The default implementation is for VoidTransform.
+    """
+
+    _EPOCH = datetime.utcfromtimestamp(0)
+
+    @staticmethod
+    def _human_day(day_ordinal):
+        time = Transform._EPOCH + timedelta(days=day_ordinal)
+        return "{0:0=4d}-{1:0=2d}-{2:0=2d}".format(time.year, time.month, time.day)
+
+    @staticmethod
+    def _unscale_decimal(decimal_value: Decimal):
+        value_tuple = decimal_value.as_tuple()
+        return int(
+            ("-" if value_tuple.sign else "")
+            + "".join([str(d) for d in value_tuple.digits])
+        )
+
+    def __init__(self, transform_string: str, repr_string: str):
+        self._transform_string = transform_string
+        self._repr_string = repr_string
+
+    def __repr__(self):
+        return self._repr_string
+
+    def __str__(self):
+        return self._transform_string
+
+    def apply(self, value):
+        return None
+
+    def can_transform(self, target: Type) -> bool:
```

Comment:
I think this should default to `False` because it needs to be implemented in subclasses.

Reply:
👌